src/software/gui/standalone_simulator/widgets/standalone_simulator_draw_function_visualizer.cpp

```diff
@@ -55,6 +69,16 @@ void StandaloneSimulatorDrawFunctionVisualizer::mouseReleaseEvent(QMouseEvent* e
         final_point   = Point(0, 0);
         ctrl_clicked  = false;
     }
+    else if (shift_clicked)
+    {
+        shift_clicked = false;
+    }
+    else if (r_clicked)
+    {
+        initial_point = Point(0, 0);
+        final_point   = Point(0, 0);
+        r_clicked     = false;
+    }
```

Comment:
What if we removed the if/else around setting *_clicked to false? Similarly with initial_point and final_point